src/stream/src/executor/temporal_join.rs

```diff
@@ -277,9 +281,18 @@ impl<S: StateStore, const T: JoinTypePrimitive> TemporalJoinExecutor<S, T> {
                     }
                 }
                 InternalMessage::Barrier(updates, barrier) => {
-                    prev_epoch = Some(barrier.epoch.curr);
+                    self.right_table.cache.evict();
+                    if let Some(vnodes) = barrier.as_update_vnode_bitmap(self.ctx.id) {
+                        let prev_vnodes =
+                            self.right_table.source.update_vnode_bitmap(vnodes.clone());
+                        if cache_may_stale(&prev_vnodes, &vnodes) {
+                            self.right_table.cache.clear();
+                        }
+                    }
+                    self.right_table.cache.update_epoch(barrier.epoch.curr);
```

Comment:
Is this already called in `right_table.update`?

Reply:
Thanks for pointing this out! I meant to only call `update_epoch` in the `into_stream` implementation, and I'll remove the other occurrence.